knockout.spin.js

```diff
@@ -1,41 +1,41 @@
 var ko = require("knockout");
-var $ = require("jquery");
+var objectAssign = require("object-assign");
+var Promise = require("pinkie-promise");
 var Spinner = require("spin.js");
 
 ko.bindingHandlers.spinner = {
 	init: function (element, valueAccessor, allBindings) {
 		// initialize the object which will return the eventual spinner
-		var deferred = $.Deferred();
-		element.spinner = deferred.promise();
+		element.spinner = new Promise(function (resolve) {
+			// force this to the bottom of the event queue in the rendering thread,
+			// so we can get the computed color of the containing element after other bindings
+			// (e.g. class, style) have evalutated.
+			// HACK: add some more delay as the class bindings of the parent fire asynchronously.
+			setTimeout(function () {
+				var options = {};
+				options.color = element.ownerDocument.defaultView.getComputedStyle(element, null).color;
+				objectAssign(options, ko.bindingHandlers.spinner.defaultOptions, ko.unwrap(allBindings.get("spinnerOptions")));
 
-		// force this to the bottom of the event queue in the rendering thread,
-		// so we can get the computed color of the containing element after other bindings
-		// (e.g. class, style) have evalutated.
-		// HACK: add some more delay as the class bindings of the parent fire asynchronously.
-		setTimeout(function () {
-			var options = {};
-			options.color = $(element).css("color");
-			$.extend(options, ko.bindingHandlers.spinner.defaultOptions, ko.unwrap(allBindings.get("spinnerOptions")));
-
-			deferred.resolve(new Spinner(options));
-		}, 30);
+				resolve(new Spinner(options));
+			}, 30);
+		});
 	},
 	update: function (element, valueAccessor) {
 		// when the spinner exists, pick up the existing one and call appropriate methods on it
 
 		var result = ko.unwrap(valueAccessor());
 
-		element.spinner.done(function (spinner) {
+		element.spinner.then(function (spinner) {
 			var isSpinning = result;
 			if (isSpinning) {
-				$(element).show();
+				element.style.display = "";
```

Comment:
Will this cause problems if the element has an inline `display` style defined? Or was jquery already overwriting it in that scenario anyway?

cc @mdarens

Reply:
The latter I believe. I can see an edge case where we would need to check for & store any inline value for the `display` property at the beginning, but in that case we already aren't handling it.

Reply:
Hi, thanks for reviewing!  What can I do to handle the `display` property edge case, if needed?

Reply:
Since we already aren't handling this edge case, and this PR wouldn't cause a regression, I have no problem merging & opening a separate issue for that.